File: engine/src/main/java/com/terralite/engine/time/GameClockSystem.java
package com.terralite.engine.time;

import com.terralite.engine.simulation.SimulationSystem;
import com.terralite.engine.simulation.SimulationTick;

import java.util.Objects;

public final class GameClockSystem implements SimulationSystem {
    private final GameClock clock;

    public GameClockSystem(GameClock clock) {
        this.clock = Objects.requireNonNull(clock, "clock");
    }

    @Override
    public void tick(SimulationTick tick) {
        clock.advance(tick.delta());
    }
}

File: engine/src/commonMain/kotlin/com/terralite/engine/time/GameClockSystem.kt
package com.terralite.engine.time

import com.terralite.engine.simulation.SimulationSystem
import com.terralite.engine.simulation.SimulationTick

class GameClockSystem(private val clock: GameClock) : SimulationSystem {
    override fun tick(tick: SimulationTick) {
        clock.advance(tick.delta)
    }
}
